src/sentry/search/events/datasets/discover.py

```diff
@@ -995,6 +1007,54 @@ def _resolve_apdex_function(self, args: Mapping[str, str], alias: str) -> Select
 
         return Function("apdex", function_args, alias)
 
+    def _resolve_web_vital_function(
+        self, args: Mapping[str, str | Column], alias: str
+    ) -> SelectType:
+        column = args["column"]
+        quality = args["quality"].lower()
+
+        if column.subscriptable != "measurements" or column.key not in VITAL_THRESHOLDS:
+            raise InvalidSearchQuery(f"count_web_vitals doesn't support {column.key}")
+        if quality not in ["good", "meh", "poor"]:
+            raise InvalidSearchQuery("Quality for count_web_vitals must be good, meh or poor")
```

Comment:
Nit: you could structure this like the following to avoid checking it twice

```python
if quality == "good":
  return blah
elif quality == "meh":
  return blah
elif quality == "poor":
  return blah

raise InvalidSearchQuery("...")
```